### ISSUE TITLE
resnet50 pretrained model has top1 ACC=79.02% ?

### ISSUE THREAD

**Issue by yangydeng** [szq0214/MEAL-V2#4]
Hi, I'm extremely interested with your work.
But I'm confuse that your pretrained Resnet50 model already have top1 Acc=79.02%, which has a big gap from your paper baseline 76.5%. (The test code also use test.py in your porject)
Have you try the pretrained model? Or did I go wrong?
Thank you.

(Resnet50 pretrained weight download from timm link: https://github.com/rwightman/pytorch-image-models/releases/download/v0.1-weights/resnet50_ram-a26f946b.pth)

**Comment by szq0214** [comment]
Hi @yangydeng,
1. Our initialization has such performance but if you run the code, you will notice that the good initialization will be damaged to some extent by the relatively large initial learning rate (0.01), so the first epoch only has ~76% accuracy, our method can recover it from 76% to 80.67%. I'm not sure how much information is still kept in the final trained network but the main purpose using good initialization is to speed up training. Even all the information in initialization is survived, our method still obtains 1.6% improvement under such already high and competitive performance of the base networks. 
2. I have some preliminary experiments in the paper to check the degree of which the good initialization may impact the final performance. But undoubtedly, more exploratory experiments can be done to study this point more thoroughly.
3. More importantly, we did not say in the paper 76.5% is our baseline result. This is the pytorch official model's performance and we list it there only for reference purposes, the training settings are different. We only stated that **our 80.67% should be a baseline for knowledge distillation**.

**Comment by yangydeng** [comment]
Hi @szq0214 ，
Thanks for your reply. 
Yes, 80%+ for resnet50 must be a breakthrough no matter it comes from 76% or 79%，and I think this work will be of great help to my current work.

**Comment by MingSun-Tse** [comment]
Hi @szq0214 ,

Thanks for your great work! Remarkable performance. I have a similar question: If the proposed method is applied to a resnet50 student _initialized from scratch_ (which should be the more usual KD setting), will it get the comparable top-1 accuracy of 80%+?

Second question: why weight decay is not used in your training? Is there any consideration behind this choice? 

Thanks again!

**Comment by szq0214** [comment]
Hi @MingSun-Tse,
1. I need to test with random initialization before I can give you any conclusion. I will find some time in the near future to do it. From my experience, I think if using larger initial lr and more training epochs, I guess the accuracy may reach to a similar value or within a reasonable gap.
2. Essentially, weight decay is doing the same thing as L2 regularization. Since L2 will penalize the large parameters in a network, in my opinion, such an operation will impair the capacity of a network. Why previous models need weigh decay and it  also seems very helpful? I think most of the previous networks are not yet saturated even those are trained with massive data aug and long-time training, so weight decay can help to prevent from overfitting and learn more information, the lost capacity is negligible and not so necessary. But in our model, the performance is already extremely high and I guess it is very close to the upper bound of the network itself's capability, so the loss of capacity will be crucial for a network and weight decay may be harmful. Moreover, since our supervision from strong teacher ensemble is very precise, the student should have enough capability to mimic such distribution, weight decay may not be applicable since it will reduce the complexity of a network.

**Comment by MingSun-Tse** [comment]
Hi @szq0214, great thanks for your reply! Looking forward to your new results. If possible, I am also curious about how MEAL_v2 compares with another sota method [CRD](https://github.com/HobbitLong/RepDistiller) on ImageNet (teacher resnet34, student resnet18, torchvision models, student randomly initialized). Greatly appreciate it if you could also have that comparison. Thanks!

**Comment by wangxianliang** [comment]
> Hi @yangydeng,
> 
> 1. Our initialization has such performance but if you run the code, you will notice that the good initialization will be damaged to some extent by the relatively large initial learning rate (0.01), so the first epoch only has ~76% accuracy, our method can recover it from 76% to 80.67%. I'm not sure how much information is still kept in the final trained network but the main purpose using good initialization is to speed up training. Even all the information in initialization is survived, our method still obtains 1.6% improvement under such already high and competitive performance of the base networks.

If the initial learning rate is not set as 0.01, but a smaller value, will the performance of the distillation model still reach 80.67%?

**Comment by szq0214** [comment]
@MingSun-Tse, a comparison of using weight decay = 1e-4 and ours (without it).

![test_top1_w_wo_wd](https://user-images.githubusercontent.com/3794909/96838003-43d78a80-1415-11eb-84a4-05ae6ab156ef.png)

**Comment by MingSun-Tse** [comment]
Hi @szq0214 , great thanks for your update! The comparison looks amazing: removing weight decay brings quite a big accuracy bump, especially in the first learning rate stage. Looking forward to the results of randomly initializing students.

**Comment by zimenglan-sysu-512** [comment]
> Hi @szq0214 , great thanks for your update! The comparison looks amazing: removing weight decay brings quite a big accuracy bump, especially in the first learning rate stage. Looking forward to the results of randomly initializing students.

@szq0214 has any updates?
